test/xml/test_c14n.rb

```diff
@@ -207,6 +239,21 @@ def test_wrong_params
         assert_raises(TypeError) { doc.canonicalize(nil, :wrong_type) }
         doc.canonicalize(nil, nil, :wrong_type)
       end
+
+      def test_multibyte_unicode
+        # https://github.com/sparklemotion/nokogiri/issues/2410
+        doc = Nokogiri.XML(%{<foo>𡏅</foo>}, nil, "EUC-JP")
+
+        # I do not understand what's going on here
+        expected = if Nokogiri.jruby?
+          %{<foo>𡏅</foo>}
+        else
+          %{<foo>陝</foo>}
```

Comment:
:four_leaf_clover:

Reply:
Yeah, I really have no idea why this is different. Maybe @AlexSun1995 has some insight, since he reported this in #2410